db/index.go

```diff
@@ -321,36 +342,66 @@ func (i *collectionUniqueIndex) newUniqueIndexError(
 	return NewErrCanNotIndexNonUniqueFields(doc.ID().String(), kvs...)
 }
 
+func (i *collectionUniqueIndex) getDocumentsIndexRecord(
+	doc *client.Document,
+) (core.IndexDataStoreKey, []byte, error) {
+	key, err := i.getDocumentsIndexKey(doc)
+	if err != nil {
+		return core.IndexDataStoreKey{}, nil, err
+	}
+	if hasIndexKeyNilField(&key) {
+		key.FieldValues = append(key.FieldValues, []byte(doc.ID().String()))
```

Comment:
question:  This looks strange, why are you doing it?

Reply:
if an index key has a nil value we tread the key as the same way as non-unique index, i.e. append docID

Reply:
Why is this not done within `getDocumentsIndexKey`?  In the current setup `getDocumentsIndexKey` appears to be returning an incorrect key.

Reply:
`getDocumentsIndexKey` is called on `collectionBaseIndex`. It doesn't return incorrect key, it returns incomplete key that is being complete in this function.

Reply:
If I have a `collectionUniqueIndex` and I want to get the key for a document, and I see a function called `getDocumentsIndexKey` I'm going to call it, and I'd expect a complete, valid key (or an error) returned.

Does `getDocumentsIndexKey` have a `WARNING` in the documentation that says it returns incomplete keys?